enterprise/backend/src/metabase_enterprise/advanced_permissions/driver/impersonation.clj

```diff
@@ -15,17 +18,43 @@
 
 (set! *warn-on-reflection* true)
 
-(defn- connection-impersonation-role
+(defn- enforce-impersonation?
+  "Takes the permission set for each group a user is in, and an impersonation policy, and determines whether the policy
+  should be enforced. This is done by checking whether the union of permissions in all *other* groups provides full
+  data access to the database. If so, we don't enforce the policy, because theo ther groups' permissions supercede it."
+  [group-id->perms-set {group-id :group_id, db-id :db_id}]
+  (let [perms-set (->> (dissoc group-id->perms-set group-id)
+                       (vals)
+                       (apply set/union))]
```

Comment:
I'm sure you've done this right, but it wasn't obvious from the names that `group-id->perms-set` maps IDs to seqs of permission sets. But since you then `(apply set/union)` to them that's how it must be

Reply:
No, it's just a map from group IDs to sets. `set/union` is being applied to the seq of sets returned by `vals`.

(just pushed a small refactor to `enforce-impersonation?`)